mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,49 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      var maxIdx = 0
+      var maxValue = if(indices(0) != 0) 0.0 else values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values incase all active node values are negative
+      if(size != values.size && maxValue < 0){
```

Comment:
This will fail for this terrible edge case.

    val a = new SparseVector(5, Array(0, 3), Array(-1.0, 0.0))
    a.argmax
    Int = 3

where actually it is 1.